### PR TITLE
[AIRFLOW-2571] airflow backfill progress bar

### PR DESCRIPTION
**Pull request by milton0825**
When you run `airflow backfill`, there is a progress bar like this:
screen shot 2018-06-07 at 2 46 22 pm

the progress bar will always stay in the bottom.

### CODE REVIEW

**Comment by milton0825** [comment]
@lyft/dp-tools-viz

File: airflow/models.py [hunk 1]
```diff
@@ -2864,6 +2865,34 @@ def normalize_schedule(self, dttm):
 
         return dttm
 
+    def get_run_dates(self, start_date, end_date=None):
```
Comment (astahlman) [inline airflow/models.py]: Curious why we need a new method for this? The BackfillJob has to already have this information in order to execute the backfill, right? Looks like we already have a list of `_dag_runs` and a counter of the number that are finished: https://github.com/lyft/incubator-airflow/blob/a52489de3f402ddf72cb6f20d6b59965bc2d668b/airflow/jobs.py#L2032-L2039

Comment (milton0825) [reply airflow/models.py]: You are right. I think we don't need this method anymore.

**Comment by milton0825** [comment]
@ArgentFalcon haven't try it yet. what is your concern tho.

**Comment by ArgentFalcon** [comment]
I background it, and the progress bar still pops up. The library you use might handle that though

**Comment by astahlman** [comment]
:+1: From me, once @ArgentFalcon's concern is addressed

**Comment by milton0825** [comment]
@ArgentFalcon other logs pops up too when I background it. Do we want to fix all of them in this PR?

**Comment by ArgentFalcon** [comment]
Python has a function to check whether you are connected to a tty, which is an option

**Comment by milton0825** [comment]
oh good to know that

**Comment by ArgentFalcon** [comment]
`os.isatty`

**Comment by feng-tao** [comment]
it seems this pr has drawed attention from airflow committers. Are you going to wait for upstream merged then cherry-picked?

**Comment by milton0825** [comment]
@feng-tao I want to gather feedbacks from both commiters and Lyft and then cherry-pick it to this PR.

**Comment by milton0825** [comment]
@lyft/dp-tools-viz

**Comment by feng-tao** [comment]
👍

**Pull request merged**